test/accessibility.js

```diff
@@ -0,0 +1,437 @@
+      var title = this.element.querySelector('h1');
+
+      title.textContent = 'really long title really long title really long title';
+      otherButton.textContent = 'another button';
+      this.element.appendChild(otherButton);
+    });
+  });
+
+  suite('Title centering:', function() {
+    teardown(function(done) {
+      accessibility.check(this.dom).then(done, done);
+    });
+
+    test('It centers the header between the surrounding buttons', function(done) {
+      this.dom.innerHTML = `<gaia-header action="menu">
+        <h1>Header title</h1>
+        <button>Foo</button>
+        <button>Bar</button>
+      </gaia-header>`;
+
+      var el = this.dom.firstElementChild;
+      var h1 = el.querySelector('h1');
+
+      afterNext(el, 'runFontFit').then(() => accessibility.check(this.dom)
+      ).then(() => {
+        this.dom.innerHTML = `<gaia-header action="menu">
+          <h1>Header title</h1>
+        </gaia-header>`;
+
+        el = this.dom.firstElementChild;
+        h1 = el.querySelector('h1');
+
+        return afterNext(el, 'runFontFit');
+      }).then(done, done);
+    });
+
+    test('It doesn\'t center if there is not enough space', function(done) {
+      this.dom.innerHTML = `<gaia-header action="menu">
+        <h1>This title is far far far far far far far far far far far far
+        far far too long to center</h1>
+        <button>Foo</button>
+        <button>Bar</button>
+      </gaia-header>`;
+
+      afterNext(this.dom.firstElementChild, 'runFontFit').then(done, done);
+    });
+
+    test('It doesn\'t have start/end padding if title is centerable', function(done) {
+      this.dom.innerHTML = `<gaia-header action="menu">
+        <h1>Title can be center</h1>
+        <button>Foo</button>
+      </gaia-header>`;
+
+      afterNext(this.dom.firstElementChild, 'runFontFit').then(done, done);
+    });
+
+    test('It should apply 10px paddings when overflowing and no buttons before/after',
+      function(done) {
+      this.dom.innerHTML = `<gaia-header>
+        <h1>This title is far far far far far far far far far far far
+        far far far far far far far too long to center</h1>
+      </gaia-header>`;
+
+      afterNext(this.dom.firstElementChild, 'runFontFit').then(done, done);
+    });
+  });
+
+  test('It works with multiple headers', function(done) {
+    this.dom.innerHTML = `<gaia-header action="back">
+      <h1 class="title-1">Title 1</h1>
+      <h1 class="title-2" hidden>Title 2</h1>
+      <button>Foo</button>
+      <button>Bar</button>
+    </gaia-header>`;
+
+    var el = this.dom.firstElementChild;
+    var title1 = el.querySelector('.title-1');
+    var title2 = el.querySelector('.title-2');
+
+    afterNext(el, 'runFontFit').then(() => accessibility.check(this.dom)
+    ).then(() => {
+      this.fontFit.reset();
+      title1.hidden = true;
+      title2.hidden = false;
+      return afterNext(el, 'runFontFit');
+    }).then(done, done);
+  });
+});
```

Comment:
In terms of an actual error message here, accessibility test-utils module will log the location and an actual error that has happened [1] (including which rule is broken).

1. https://github.com/gaia-components/test-utils/blob/master/src/accessibility.js#L19-L25